core/tests/protractor_utils/ExplorationEditorFeedbackTab.js

```diff
@@ -0,0 +1,110 @@
+  /*
+   * Interactive elements
+   */
+  var explorationFeedbackSubject = element(
+    by.css('.protractor-test-exploration-feedback-subject'));
+  var feedbackTabRow = element(
+    by.css('.protractor-test-oppia-feedback-tab-row'));
+  var explorationFeedback = element(
+    by.css('.protractor-test-exploration-feedback'));
+  var feedbackBackButton = element(
+    by.css('.protractor-test-oppia-feedback-back-button'));
+  var feedbackResponseTextArea = element(
+    by.css('.protractor-test-feedback-response-textarea'));
+  var suggestionRowClassName = '.protractor-test-oppia-feedback-tab-row';
+  /*
+   * Buttons
+   */
+  var acceptSuggestionButton = element(
+    by.css('.protractor-test-exploration-accept-suggestion-btn'));
+  var feedbackSendResponseButton = element(
+    by.css('.protractor-test-oppia-feedback-response-send-btn'));
+  var viewSuggestionButton = element(
+    by.css('.protractor-test-view-suggestion-btn'));
+  /*
+   * Workflows
+   */
+  this.acceptSuggestion = function(suggestionDescription) {
+    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
+      var matchingSuggestionRows = rows.filter(function() {
+        return explorationFeedbackSubject.getText().then(function(subject) {
+          return suggestionDescription.indexOf(subject) !== -1;
+        });
+      });
+
+      browser.waitForAngular();
+      matchingSuggestionRows[0].click();
+      general.waitForSystem();
+      viewSuggestionButton.isDisplayed().then(function(){
+        viewSuggestionButton.click();
+      });
+      general.waitForSystem();
+      acceptSuggestionButton.isDisplayed().then(function(){
+        acceptSuggestionButton.click();
+      });
+    });
+  };
+
+  this.expectToHaveFeedbackThread = function() {
+    expect(feedbackTabRow.isPresent()).toBe(true);
+  };
+
+  this.getSuggestionThreads = function() {
+    var threads = [];
+    browser.waitForAngular();
+    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
+      rows.forEach(function() {
+        explorationFeedbackSubject.getText().then(function(subject) {
+          threads.push(subject);
+        });
+      });
+      return threads;
+    });
+  };
+
+  this.readFeedbackMessages = function() {
+    var messages = [];
+
+    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
+      rows.forEach(function(row) {
+        row.click();
+        explorationFeedback.getText().then(function(message) {
+          messages.push(message);
+        });
+        feedbackBackButton.click();
+      });
+      return messages;
+    });
+  };
+
+  this.sendResponseToLatestFeedback = function(feedbackResponse) {
+    element.all(by.css('.protractor-test-oppia-feedback-tab-row')).
+      first().click();
+    feedbackResponseTextArea.sendKeys(feedbackResponse);
+    feedbackSendResponseButton.click();
+  };
+};
+exports.ExplorationEditorFeedbackTab = ExplorationEditorFeedbackTab;
```

Comment:
I would add a blank line above this one. Ditto in other files.

Reply:
Done!